### DISCUSSION TITLE
Is there a way to validate a payload in emqx?

### DISCUSSION THREAD

**Discussion by ZangoTL** [emqx/emqx#13029] [Q&A] [answered]
I want to know if there is any way to validate a payload before it was published to a topic in emqx. I cant find anything in the docs. The rules engine looked promising for this purpose at first but i dont think it fits as neetly as i would it to be.

Thank you in advance for any clarification

**Comment by ieQu1** [comment]
I don't think rule engine can prevent messages from being routed by the broker. However, such filtering can be done by a custom plugin: plugins can hook into [various broker flows](https://www.emqx.io/docs/en/v5.0/extensions/hooks.html#hookpoint-list) and they can access all message fields, including the payload. Dropping messages based on the payload can be done in a similar way the authorization plugins check client permissions.

**Comment by id** [comment, marked as answer]
[Message validation EIP](https://github.com/emqx/eip/blob/main/active/0024-message-validation.md) will be implemented in the upcoming 5.7.0 release. It will be an Enterprise edition feature.
